connectors/aws-cert/src/main/java/com/vmware/connectors/aws/cert/AwsCertController.java

```diff
@@ -0,0 +1,331 @@
+                .map(this::parseCardInfoOutOfResponse)
+                .reduce(
+                        new Cards(),
+                        (cards, info) -> appendCard(cards, info, routingPrefix)
+                )
+                .toSingle()
+                .map(ResponseEntity::ok);
+    }
+
+    private List<String> validateUrls(List<String> approvalUrls) {
+        return approvalUrls.stream()
+                .filter(this::validateUrl)
+                .collect(Collectors.toList());
+    }
+
+    private boolean validateUrl(String approvalUrl) {
+        try {
+            UriComponents uriComponents = UriComponentsBuilder.fromHttpUrl(approvalUrl).build();
+            return verifyHost(uriComponents) && verifyPath(uriComponents);
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+    }
+
+    private boolean verifyHost(UriComponents uriComponents) {
+        String host = uriComponents.getHost().toLowerCase(Locale.US);
+        return host.equals(certificateApprovalHost)
+                || host.endsWith(certificateApprovalHost)
+                && host.charAt(host.lastIndexOf(certificateApprovalHost) - 1) == '.';
+    }
+
+    private boolean verifyPath(UriComponents uriComponents) {
+        return uriComponents.getPath().equals(certificateApprovalPath);
+    }
+
+    private Observable<Pair<String, ResponseEntity<String>>> getAllCardsInfo(Set<String> approvalUrls) {
+        return Observable
+                .from(approvalUrls)
+                .flatMapSingle(this::callForCardInfo);
+    }
+
+    private Single<Pair<String, ResponseEntity<String>>> callForCardInfo(String approvalUrl) {
+        logger.trace("callForCardInfo called: approvalUrl={}", approvalUrl);
+
+        ListenableFuture<ResponseEntity<String>> response = rest.exchange(
+                UriComponentsBuilder
+                        .fromHttpUrl(approvalUrl)
+                        .build()
+                        .toUri(),
+                HttpMethod.GET,
+                HttpEntity.EMPTY,
+                String.class
+        );
+
+        return Async.toSingle(response)
+                .map(responseEntity -> Pair.of(approvalUrl, responseEntity));
+    }
+
+    private AwsCertCardInfo parseCardInfoOutOfResponse(Pair<String, ResponseEntity<String>> pair) {
+        String approvalUrl = pair.getLeft();
+        logger.trace("parseCardInfoOutOfResponse called: approvalUrl={}", approvalUrl);
+
+        String html = pair.getRight().getBody();
+        AwsCertCardInfo info = new AwsCertCardInfo();
+
+        Document doc = Jsoup.parse(html);
+        Elements elements = doc.body().children();
+        Elements rows = elements.select("table > tbody > tr");
+
+        rows.forEach(row -> {
+            String label = row.child(0).text().toLowerCase(Locale.US);
+            String value = row.child(1).text();
+            if (label.contains("domain")) {
```

Comment:
These are a quick and dirty attempt at fuzzy matching in the hopes that small changes to the approval page's html won't break this connector.